design/ceph/ceph-telemetry.md

```diff
@@ -0,0 +1,116 @@
+    "version"
+  }
+}
+```
+
+### Overall metrics
+- `rook-version` - Rook version.
+  - **DISCUSS:** should we add `rook/version` or rename this to maintain a consistent hierarchy?
+- `rook/kubernetes/version` - Kubernetes version.
+  - Ceph already collects os/kernel versions.
+- `rook/kubernetes/node-count` - Number of Kubernetes nodes.
+- `rook/csi/version` - Ceph CSI version.
+- Number of CSI plugin pods running - This gives idea of how many k8s nodes are in the k8s cluster that
+  are allowed to use Ceph storage.
+  - It is easiest count plugin pods separately since each DaemonSet is a different label. Plus, it
+    can show if some CSI drivers are ever disabled.
+  - `rook/csi/rbd-plugin-pod-count`
+  - `rook/csi/cephfs-plugin-pod-count`
+  - `rook/csi/nfs-plugin-pod-count`
+
+### CephCluster metrics
+- `rook/cluster/nodes-with-ceph-pods-count` - Number of nodes on which the cluster has running pods.
+  - Since clusters with portable PVCs have one "node" per PVC, this will help show the actual node
+    count for Rook installs in portable environments
+  - We can get this by counting the number of crash collector pods; however, if any users disable
+    the crash collector, we should report an unknown value, possibly `unknown`.
+- `rook/cluster/storage-class/count` - Number of storage classes related to the current Ceph cluster. We can
+  also break this up by specific kinds of storage classes, but the overall may be useful if new
+  classes are added in the future.
+  - `rook/cluster/storage-class/rbd-count`
+  - `rook/cluster/storage-class/cephfs-count`
+  - `rook/cluster/storage-class/nfs-count`
+  - `rook/cluster/storage-class/bucket-count`
+- `rook/cluster/storage-class-device-set/count` - Total number of storage class device sets.
+- `rook/cluster/storage-class-device-set/portable-count` - Number of storage class device sets that
+  are configured to be portable.
+- `rook/cluster/storage-class-device-set/non-portable-count` - Number of storage class device sets
+  that are not configured to be portable.
+- `rook/cluster/mons/are-pvc` - true/false whether mons are on PVC.
+  - Leave some ability to expand the mon section in the future if desired.
+- `rook/cluster/network-provider` - The network provider used for the cluster. e.g., default, host, multus.
+- `rook/cluster/external-mode` - true/false if the cluster is in external mode.
+
+## Information Rook is interested in that is already collected by Ceph telemetry
+- RBD pools (name is stripped), some config info, and the number of them.
+- Ceph Filesystems and MDS info.
+- RGW count, zonegroups, zones.
+- RBD mirroring info.
+
+## Metrics that Rook would like to see added to Ceph's telemetry
+- Number of NFS clusters
+  - Ceph can count NFS clusters by listing the namespaces in the `.nfs` pool and report those
+    namespaces that have `conf-nfs.<namespace>` objects present
+- Filesystem mirroring info
+- RGW bucket notification usage (Yuval may have good detail on this point)
```

Comment:
Any specific data points would be helpful to know here (understood that this is a WIP design concept though)

Reply:
@BlaineEXE How about if we track follow-up additions as issues? I'm afraid it will get lost in this doc. Issues at least remind us occasionally to do something from the stale bot.